service/siteService.js

```diff
@@ -17,27 +17,31 @@ module.exports = {
 		const cachePath = cacheService.getCachePath(requestPath);
 		const cachedResult = cacheService.readCache(cachePath);
 
-		if (cachedResult != null) {
-			console.log("Cache hit. Returning cached result for " + requestPath);
-			return cachedResult
-		} else {
-			const siteRecords = await airtableClient.getMutualAidSites()
-				.catch( e => {
-					console.error("There was an error getting mutual aid sites: " + e.message)
-					// TODO: Send slack? alert so there's visibility into the error!!
-					return cacheService.readCacheBypassInterval(cachePath)
-				})
-			const result = siteRecords
-					.filter(validateRecord)
-					.map(mapRecordFields)
-			cacheService.writeCache(cachePath, result);
-			return result
+		try {
+			if (cachedResult != null) {
+				console.log("Cache hit. Returning cached result for " + requestPath);
+				return cachedResult
+			} else {
+				const siteRecords = await airtableClient.getMutualAidSites()
+					.catch( e => {
+						console.error("There was an error getting mutual aid sites: " + e.message)
+						// TODO: Send slack? alert so there's visibility into the error!!
+						return cacheService.readCacheBypassInterval(cachePath)
```

Comment:
I realize this isn't one of the files you changed (I'll take the L on missing it in the original review), but rather than making it a separate function, it may be cleaner to pass it as a flag to `readCache()`, so the function header looks like `readCache( path, ignoreFreshness)` or something.  Two reasons:

* As a caller, you only need to worry about one function header.
* You leverage the same error checking behavior that already exists.  Since the method used here currently skips checking file existence, I don't know what happens if you perform `fs.readFile()` on a file that doesn't exist

Reply:
mmm that's a good point.
kind of like ```
  readCache: function(path, ignoreFreshness) {
    var shouldSendCache = false;

    if (fs.existsSync(path)) {
      if (ignoreFreshness) {
        return JSON.parse(fs.readFileSync(path, 'utf8'));
      }
      var cachedTime = fs.statSync(path).ctime;

      if ((new Date().getTime() / 1000.0 - cachedTime / 1000.0) < cacheInterval) {
        shouldSendCache = true;
      }
    }

    if (!shouldSendCache) return null;
    else return JSON.parse(fs.readFileSync(path, 'utf8'));

  }
```
??
(probably could be more elegantly written)